components/net/http_cache.rs

```diff
@@ -43,6 +43,12 @@ impl CacheKey {
             url: request.url().clone(),
             request_headers: request.headers
                                       .iter()
+                                      .filter(|header| {
+                                          match header.name().to_lowercase().as_ref() {
```

Comment:
I don't think you need the `as_ref` call here if you have the `ref` keyword in the line above for `header`.

Reply:
I tried that, and also putting a & in front of `header.name()`, and both would give 
```
expected type `&std::string::String`
              found type `&'static str`
```